services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/models/V6MatchCode.java

```diff
@@ -0,0 +1,147 @@
+package com.mapbox.api.geocoding.v6.models;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.api.geocoding.v6.V6GeocodingAdapterFactory;
+
+/**
+ * The {@link V6MatchCode} object in the Geocoding API helps you understand
+ * how the resulting address feature aligns with the query submitted.
+ * Available only for address-type features, the {@link V6MatchCode} provides a breakdown
+ * of how each element of the result matches with the query, plus an overall Confidence score,
+ * based on how well it matches. This can help you make decisions about what results to keep
+ * or throw out based on your application's tolerance for fuzzy matching on the query.
+ * <p>
+ * Smart Address Match is available for all forward geocoding requests that return
+ * an address type feature. It works best when using Structured Input forward queries,
+ * as the request components must be typed explicitly.
+ *
+ * @see <a href="https://docs.mapbox.com/api/search/geocoding-v6/#smart-address-match">Smart Address Match</a>
+ */
+@AutoValue
+public abstract class V6MatchCode extends V6Object {
+
+  /**
+   * Match code for address number.
+   *
+   * @return match code for address number
+   */
+  @Nullable
+  @SerializedName("address_number")
+  public abstract String addressNumber();
+
+  /**
+   * Match code for street.
+   *
+   * @return match code for street
+   */
+  @Nullable
+  @SerializedName("street")
+  public abstract String street();
```

Comment:
What's a "match code for street"? Or for locality, place, etc.? Do you have an example?

Reply:
https://docs.mapbox.com/api/search/geocoding-v6/#smart-address-match

When API will be public I'll make @StringDef for this.